src/go/k8s/pkg/resources/statefulset_scale.go

```diff
@@ -428,10 +450,30 @@ func setCurrentReplicas(
 	}
 
 	log.Info("Scaling StatefulSet", "replicas", replicas)
-	pandaCluster.Status.CurrentReplicas = replicas
-	if err := c.Status().Update(ctx, pandaCluster); err != nil {
-		return fmt.Errorf("could not scale cluster %s to %d replicas: %w", pandaCluster.Name, replicas, err)
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
i'm seeing often issues with conflicts in this update. now using retry as we should

Reply:
Rather than using retry why not use a patch here? Patches, by default, won't check resource version and will only change the value that you've changed since the last read (Assuming you're using a client side patch).

`.Status` is the one exception from my preference to rely on resource version checking. You've already done the work, so you're the one that know the actual state of the status.  Statuses, despite being a subresource, still rely on the checking the resource version of the whole object. So you might be getting conflicts due to someone making a change to the Spec which doesn't impact your change on the status.